### PR TITLE
add partialFilterExpression support for create_index

### LINKED ISSUE #687: createIndex don't support partialFilterExpression
I noticed that the partialFilterExpression argument for [createIndex](https://docs.mongodb.com/manual/core/index-partial/#create-a-partial-index) isn't respected for mongomock.

```python
db.users.createIndex(
   { name: 1, accountId: 1 },
   { partialFilterExpression: { accountId: { $exists: true } } } # this argument is ignored by mongomock
)
```

### PR DESCRIPTION
**Pull request by chdsbd**
fixes #687

### CODE REVIEW

File: tests/test__collection_api.py [hunk 1]
```diff
@@ -1645,6 +1645,27 @@ def test_sparse_unique_index_dup(self):
 
         self.assertEqual(self.db.collection.find({}).count(), 1)
 
+    def test_unique_index_partial_filter_expression(self):
```
Comment (pcorpet) [inline tests/test__collection_api.py:1648]: Great test, can you try to add the same in test__mongomock to ensure we are consistent with `pymongo + MongoDB` please?

**Review by pcorpet — CHANGES_REQUESTED** [review]
Please fix lint errors detected by the CI.

File: tests/test__collection_api.py (hunk 1 above)
Comment (pcorpet) [reply tests/test__collection_api.py:1648]: Sorry I wasn't clear enough: please add the equivalent test, in a `_CollectionComparisonTest` test suite. Calls should use `self.cmp.do` and the expectations should not the be on the actual result but that the results match between mongomock and pymongo. I hope that it's clearer.

**Comment by jschlyter** [comment]
Any updates on this PR?

**Comment by Naushi** [comment]
Any updates?

File: tests/test__collection_api.py (hunk 1 above)
Comment (pcorpet) [reply tests/test__collection_api.py:1648]: This is still unresolved

**Comment by pcorpet** [comment]
There are remaining unresolved comments. @Naushi @jschlyter you can create a concurrent PR if you want to finish the cleanup.

**Comment by pcorpet** [comment]
#784 was created, and I'm reviewing there. Closing this one.

**Pull request closed without merging**